coral/src/app/features/topics/overview/messages/TopicMessages.tsx

```diff
@@ -17,38 +23,69 @@ function isNoContentResult(
 
 function TopicMessages() {
   const { topicName } = useTopicDetails();
+  const [offset, setOffset] = useOffsetFilter();
   const {
     data: consumeResult,
     isLoading,
     isError,
+    isRefetching,
     error,
+    refetch: updateResults,
   } = useQuery({
-    queryKey: ["topicMessages"],
+    queryKey: ["topicMessages", offset],
     queryFn: () =>
       getTopicMessages({
         topicName,
         consumerGroupId: "notdefined",
         envId: "2",
-        offsetId: "5",
+        offsetId: offset,
       }),
-    keepPreviousData: true,
+    keepPreviousData: false,
   });
 
-  if (isNoContentResult(consumeResult)) {
-    return (
-      <EmptyState title="No messages">
-        No Message matched your criteria.
-      </EmptyState>
-    );
+  const isConsuming = isLoading || isRefetching;
+
+  function handleUpdateResultClick(): void {
+    updateResults();
+  }
+
+  function handleOffsetChange(offset: Offset): void {
+    setOffset(offset);
   }
 
   return (
     <TableLayout
-      filters={[]}
-      isLoading={isLoading}
+      filters={[
+        <TopicMessageOffsetFilter
+          key={"offset"}
+          value={offset}
+          disabled={isConsuming}
+          onChange={handleOffsetChange}
+        />,
+        <Box.Flex key={"consume"} justifyContent="flex-end">
+          <Button.Primary
+            onClick={handleUpdateResultClick}
+            disabled={isConsuming}
+            loading={isConsuming}
+            aria-label={`Consume and display the latest ${offset} messages from topic ${topicName}`}
+            icon={refreshIcon}
+          >
+            Update results
+          </Button.Primary>
+        </Box.Flex>,
+      ]}
+      isLoading={isConsuming}
       isErrorLoading={isError}
       errorMessage={error}
-      table={<TopicMessageList messages={consumeResult ?? {}} />}
+      table={
+        isNoContentResult(consumeResult) ? (
+          <EmptyState title="No messages">
+            This Topic contains no Messages.
+          </EmptyState>
+        ) : (
+          <TopicMessageList messages={consumeResult ?? {}} />
```

Comment:
There is a header on every tab in the design, and Subscriptions implements it. I think we want to add it here as well ^^ 

<img width="968" alt="Screenshot 2023-05-26 at 09 36 40" src="https://github.com/aiven/klaw/assets/20607294/6b1948df-82e2-4033-ad73-ab3bafd242c4">

Reply:
Thanks, somehow i missed that. Thank you 🙂